### ISSUE TITLE
Segmentation fault when resuming from suspend when lockscreen or notifications are enabled

### ISSUE THREAD

**Issue by rrueger** [swaywm/sway#7606]
- Sway version: `1.9-dev-48d6eda3 (May 24 2023, branch 'master')`
- OS: `archlinux`
- uname -r:  `Linux <hostname> 6.3.3-arch1-1 #1 SMP PREEMPT_DYNAMIC Sun, 21 May 2023 16:15:22 +0000 x86_64 GNU/Linux`

This is a somewhat peculiar issue, however I am able to successfully replicate it.

If either

- A notification deamon is running (I tried `dunst` and `mako`)
- The screen is locked with `swaylock`

when the computer is suspended, I get a core segmentation fault when resuming from suspend. The screen wakes, however if I do anything to cause the content to update, e.g. move the mouse, type my password for the locker, sway crashes.

However, if I remove the `include /etc/sway/config.d/50-systemd-user.conf` line from my sway config, the issue no longer appears. Regardless of locking/notifications. I isolated that `WAYLAND_DISPLAY` seems to cause the issues here.

**Comment by emersion** [comment]
Please provide a stack trace. You can do so by compiling from source, reproducing the crash and then running `coredumpctl gdb` and then `bt full`.

Here are some instructions to compile from source: https://github.com/swaywm/sway/wiki/Development-Setup#compiling-as-a-subproject

**Comment by rrueger** [comment]
It is worth noting, that I am using a Dell XPS 15 9510 with a discrete nvidia graphics card. That is why I must use `--unsupported-gpu`. The graphics card is almost always disabled. My sway session runs on my intel cpu/gpu.

**Comment by gwenhael-le-moine** [comment]
I had a similar issue on 2 machines, both running Slackware-current with sway compiled regularly from master. One Thinkpad T590 with an external screen plugged and my desktop PC (AMD 5600x with RX6600XT, 1 screen).

On the T590 sway has been crashing for quite a long time and the desktop only more recently.

I only got so far to notice that the crash occurred going out of idle, presumably when `swayidle` issue the command `swaymsg "output * power on"`.

I think this has been fixed by the last commit ( https://github.com/swaywm/sway/commit/8b4b65d66552d782d2c38a17756baf928b90e670 ), at least that's what I see in my cases.
